includes/admin/settings/class-settings-gateways.php

```diff
@@ -344,15 +347,19 @@ public function render_enabled_gateways($field, $settings)
             $current_tab = give_get_current_setting_tab();
             $current_section = give_get_current_setting_section();
 
-            $v2Gateways = give_get_ordered_payment_gateways(
-                array_intersect_key($gateways, give()->gateways->getPaymentGateways(2)),
-                2
-            );
+            $v2Gateways = array_intersect_key($gateways, give()->gateways->getPaymentGateways(2));
             $v3Gateways = give_get_ordered_payment_gateways(
                 array_intersect_key($gateways, give()->gateways->getPaymentGateways(3)),
                 3
             );
 
+            // Add legacy payment gateways to v2 gateways if they exist.
+            if ($legacyPaymentGateways = array_diff_key($gateways, $v2Gateways, $v3Gateways)) {
+                $v2Gateways = array_merge($v2Gateways, $legacyPaymentGateways);
+            }
+
+            $v2Gateways = give_get_ordered_payment_gateways($v2Gateways, 2);
+
```

Comment:
Oh I see, because some gateways are still not in the gateway api.  This is fine but i think it could be cleaned up by doing something like this (pseudo code):

```
$legacyGateways = array_filter($gateways, function($value, $key){
  return !give()->gateways->hasPaymentGateway($key);
  }, ARRAY_FILTER_USE_BOTH);
  
$v2Gateways = array_merge($legacyGateways, array_intersect_key($gateways, give()->gateways->getPaymentGateways(2)));
```

Reply:
fixed: https://github.com/impress-org/givewp/pull/6995/commits/177245d8d6c30ddccdbac3bacbe9550df104cba8